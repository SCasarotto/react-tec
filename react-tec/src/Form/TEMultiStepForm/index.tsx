//
// NOTES:
//
// 10-11-18 SC
// - May be able to be improved by moving to a render prop form but seems fine for now.
// - Could also improve by adding react-transition-group around each step
//

import React, { useState } from 'react'

import {
	Container,
	StepContainer,
	StepBar,
	StepWrapper,
	StepNumber,
	StepTitle,
	Form,
	ButtonContainer,
	StepButton,
} from './styledComponents'

export interface TEMultiStepFormStepData {
	title: string
	component: React.ReactNode
	onPrevious?(): Promise<any>
	onNext?(): Promise<any>
}
export interface TEMultiStepFormProps {
	className?: string
	handleCancelOnClick?(): void
	stepData: TEMultiStepFormStepData[]
	roundedButtons: boolean
	onSubmit(): Promise<any>
	onStepChange?(step: number): void
}
const TEMultiStepForm: React.FC<TEMultiStepFormProps> = (props) => {
	const [currentStep, setCurrentStep] = useState(0)

	const updateAndEmitStep = (step: number) => {
		const { onStepChange } = props
		setCurrentStep(step)
		if (onStepChange) {
			onStepChange(step)
		}
	}
	const handlePreviousPressed = async () => {
		const { handleCancelOnClick, stepData } = props

		const moveBack = () => {
			if (currentStep === 0) {
				if (handleCancelOnClick) {
					handleCancelOnClick()
				}
			} else {
				updateAndEmitStep(currentStep - 1)
			}
		}

		try {
			const onPrevious =
				stepData &&
				stepData[currentStep] &&
				stepData[currentStep].onPrevious
			if (onPrevious) {
				await onPrevious()
				moveBack()
			} else {
				moveBack()
			}
		} catch (e) {
			console.log(e)
		}
	}
	const handleNextPressed = async () => {
		const { onSubmit, stepData } = props

		const moveForward = async () => {
			try {
				if (currentStep + 1 === stepData.length) {
					await onSubmit()
					updateAndEmitStep(0)
				} else {
					updateAndEmitStep(currentStep + 1)
				}
			} catch (e) {
				console.log(e)
			}
		}
		try {
			const onNext =
				stepData &&
				stepData[currentStep] &&
				stepData[currentStep].onNext
			if (onNext) {
				await onNext()
				moveForward()
			} else {
				moveForward()
			}
		} catch (e) {
			console.log(e)
		}
	}

	const {
		className = '',
		handleCancelOnClick,
		stepData,
		// roundedButtons,
	} = props
	if (!stepData || (stepData && stepData.length === 0)) {
		return null
	}

	return (
		<Container className={`TEMultiStepForm ${className}`}>
			<StepContainer className="TEMultiStepFormStepContainer">
				<StepBar
					stepCount={stepData.length}
					className="TEMultiStepFormStepBar"
				/>
				{stepData.map((step, index) => {
					const { title } = step

					return (
						<StepWrapper
							key={index}
							stepCount={stepData.length}
							className="TEMultiStepFormStepWrapper">
							<StepNumber
								active={currentStep === index}
								className="TEMultiStepFormStepNumber">
								{index + 1}
							</StepNumber>
							<StepTitle className="TEMultiStepFormStepTitle">
								{title}
							</StepTitle>
						</StepWrapper>
					)
				})}
			</StepContainer>
			<Form className="TEMultiStepFormForm">
				{stepData[currentStep] && stepData[currentStep].component}
			</Form>
			<ButtonContainer className="TEMultiStepFormButtonContainer">
				{(handleCancelOnClick || currentStep > 0) && (
					<StepButton
						onClick={handlePreviousPressed}
						// position="left"
						// rounded={roundedButtons}
						className="TEMultiStepFormStepButton TEMultiStepFormStepButtonLeft">
						{currentStep === 0 ? 'cancel' : 'previous'}
					</StepButton>
				)}
				<StepButton
					onClick={handleNextPressed}
					// position="right"
					// rounded={roundedButtons}
					// singleButton={!handleCancelOnClick && currentStep === 0}
					className="TEMultiStepFormStepButton TEMultiStepFormStepButtonright">
					{currentStep + 1 === stepData.length ? 'submit' : 'next'}
				</StepButton>
			</ButtonContainer>
		</Container>
	)
}

export default TEMultiStepForm
