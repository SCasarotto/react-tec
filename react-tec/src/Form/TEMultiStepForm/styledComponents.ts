import styled from 'styled-components'

import TEForm from '../TEForm'
import TEButton from '../TEButton'

export const Container = styled.div``
export const StepContainer = styled.div`
	width: 100%;
	display: flex;
	position: relative;
	margin-bottom: 30px;

	@media (max-width: 500px) {
		margin-bottom: 15;
	}
`
export const StepBar = styled.div`
	position: absolute;
	top: 14px;
	background-color: ${(props) => props.theme.primary};
	height: 2px;
	width: ${(props) => 100 - 100 / props.stepCount}%;
	left: ${(props) => 100 / props.stepCount / 2}%;
`
export const StepWrapper = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	width: ${(props) => 100 / props.stepCount}%;
`
export const StepNumber = styled.div`
	text-align: center;
	width: 30px;
	height: 30px;
	margin-bottom: 10px;
	line-height: 30px;
	border: 1px solid ${(props) => props.theme.primary};
	z-index: 1;
	background-color: ${(props) =>
		props.active ? props.theme.primary : props.theme.white};
	color: ${(props) =>
		props.active ? props.theme.white : props.theme.primary};
	transition: color 0.2s ease-in, background-color 0.2s ease-in;
`
export const StepTitle = styled.span`
	text-align: center;
	font-size: 18px;
	display: block;

	@media (max-width: 500px) {
		font-size: 16px;
	}
	@media (max-width: 350px) {
		font-size: 14px;
	}
`

export const Form = styled(TEForm)`
	margin-bottom: 30px;
`
export const ButtonContainer = styled.div`
	display: flex;
`

//TODO: Return to this and figure out way to handle popups and panel use cases without being weird to use in general
/*
		// const { position, rounded, singleButton, theme } = props
		// let styles = `
		//     padding: 10px 20px;
		//     color: ${theme.primary};
		//     background-color: ${theme.white};
		//     border-top: 1px solid ${theme.primary};
		//     border-bottom: none;
		//     transition: border 0.2s ease-in, color 0.2s ease-in, background-color 0.2s ease-in;
		// `

		// switch (position) {
		//     case 'left':
		//         styles += `
		//                 border-left: none;
		//                 border-right: 1px solid ${theme.primary};

		//                 border-bottom-left-radius: ${rounded ? '5px' : '0px'};
		//                 border-top-right-radius: 0px;
		//                 border-top-left-radius: 0px;
		//                 border-bottom-right-radius: 0px;

		//                 :hover {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: none;
		//                     border-right: 1px solid ${theme.primary};
		//                     border-bottom: none;
		//                 }
		//                 :active {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: none;
		//                     border-right: 1px solid ${theme.primary};
		//                     border-bottom: none;
		//                 }
		//             `
		//         break
		//     case 'right':
		//         styles += `
		//                 border-left: ${singleButton ? 'none' : `1px solid ${theme.primary}`};
		//                 border-right: none;

		//                 border-bottom-right-radius: ${rounded ? '5px' : '0px'};
		//                 border-bottom-left-radius: ${singleButton && rounded ? '5px' : '0px'};
		//                 border-top-right-radius: 0px;
		//                 border-top-left-radius: 0px;

		//                 :hover {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: ${singleButton ? 'none' : `1px solid ${theme.primary}`};
		//                     border-right: none;
		//                     border-bottom: none;
		//                 }
		//                 :active {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: ${singleButton ? 'none' : `1px solid ${theme.primary}`};
		//                     border-right: none;
		//                     border-bottom: none;
		//                 }
		//             `
		//         break
		//     default:
		//         break
		// }
	*/
export const StepButton = styled(TEButton)``
