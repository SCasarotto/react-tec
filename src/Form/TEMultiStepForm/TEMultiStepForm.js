//
//TE Version 0.3.0
//

//
// NOTES:
//
// 10-11-18 SC
// - May be able to be improved by moving to a render prop form but seems fine for now.
// - Could also improve by adding react-transition-group around each step
//

import React, { useState } from 'react'
import PropTypes from 'prop-types'

import {
	Container,
	StepContainer,
	StepBar,
	StepWrapper,
	StepNumber,
	StepTitle,
	Form,
	ButtonContainer,
	StepButton,
} from './styledComponents'

const TEMultiStepForm = (props) => {
	const [currentStep, setCurrentStep] = useState(0)

	const updateAndEmitStep = (step) => {
		const { onStepChange } = props
		setCurrentStep(step)
		if (onStepChange) {
			onStepChange(step)
		}
	}
	const handlePreviousPressed = () => {
		const { handleCancelOnClick, stepData } = props

		const moveBack = () => {
			if (currentStep === 0) {
				if (handleCancelOnClick) {
					handleCancelOnClick()
				}
			} else {
				updateAndEmitStep(currentStep - 1)
			}
		}

		if (stepData && stepData[currentStep] && stepData[currentStep].onPrevious) {
			stepData[currentStep]
				.onPrevious()
				.then(() => moveBack())
				.catch((error) => console.log(error))
		} else {
			moveBack()
		}
	}
	const handleNextPressed = () => {
		const { onSubmit, stepData } = props

		const moveForward = () => {
			if (currentStep + 1 === stepData.length) {
				onSubmit()
					.then(() => {
						updateAndEmitStep(0)
					})
					.catch((error) => console.log(error))
			} else {
				updateAndEmitStep(currentStep + 1)
			}
		}

		if (stepData && stepData[currentStep] && stepData[currentStep].onNext) {
			stepData[currentStep]
				.onNext()
				.then(() => moveForward())
				.catch((error) => console.log(error))
		} else {
			moveForward()
		}
	}

	const { className, handleCancelOnClick, stepData, roundedButtons } = props
	if (!stepData || (stepData && stepData.length === 0)) {
		return null
	}

	return (
		<Container className={className}>
			<StepContainer className='TEMultiStepFormStepContainer'>
				<StepBar stepCount={stepData.length} className='TEMultiStepFormStepBar' />
				{stepData.map((step, index) => {
					const { title } = step

					return (
						<StepWrapper
							key={index}
							stepCount={stepData.length}
							className='TEMultiStepFormStepWrapper'
						>
							<StepNumber
								active={currentStep === index}
								className='TEMultiStepFormStepNumber'
							>
								{index + 1}
							</StepNumber>
							<StepTitle className='TEMultiStepFormStepTitle'>{title}</StepTitle>
						</StepWrapper>
					)
				})}
			</StepContainer>
			<Form className='TEMultiStepFormForm'>
				{stepData[currentStep] && stepData[currentStep].component}
			</Form>
			<ButtonContainer className='TEMultiStepFormButtonContainer'>
				{(handleCancelOnClick || currentStep > 0) && (
					<StepButton
						onClick={handlePreviousPressed}
						position='left'
						rounded={roundedButtons}
						className='TEMultiStepFormStepButton TEMultiStepFormStepButtonLeft'
					>
						{currentStep === 0 ? 'cancel' : 'previous'}
					</StepButton>
				)}
				<StepButton
					onClick={handleNextPressed}
					position='right'
					rounded={roundedButtons}
					singleButton={!handleCancelOnClick && currentStep === 0}
					className='TEMultiStepFormStepButton TEMultiStepFormStepButtonright'
				>
					{currentStep + 1 === stepData.length ? 'submit' : 'next'}
				</StepButton>
			</ButtonContainer>
		</Container>
	)
}

TEMultiStepForm.propTypes = {
	onStepChange: PropTypes.func,
	onSubmit: PropTypes.func.isRequired,
	stepData: PropTypes.arrayOf(
		PropTypes.shape({
			title: PropTypes.string.isRequired,
			component: PropTypes.element,
			onPrevious: PropTypes.func,
			onNext: PropTypes.func,
		}).isRequired
	).isRequired,
	roundedButtons: PropTypes.bool,
}

export default TEMultiStepForm
