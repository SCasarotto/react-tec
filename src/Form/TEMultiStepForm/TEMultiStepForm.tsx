//
// NOTES:
//
// 10-11-18 SC
// - May be able to be improved by moving to a render prop form but seems fine for now.
// - Could also improve by adding react-transition-group around each step
//

import React, { useState } from 'react'

import { TEForm } from '../TEForm'

import {
	Container,
	StepContainer,
	StepBar,
	StepWrapper,
	StepNumber,
	StepTitle,
	ButtonContainer,
	StepButton,
} from './styledComponents'

export interface TEMultiStepFormStepData {
	title: string
	component: React.ReactNode
	// TODO: Try to make these types better
	onPrevious?(): Promise<any>
	onNext?(): Promise<any>
}
export interface TEMultiStepFormProps {
	className?: string
	handleCancelOnClick?(): void
	stepData: TEMultiStepFormStepData[]
	// roundedButtons?: boolean
	// TODO: Try to make these types better
	onSubmit(): Promise<any>
	onStepChange?(step: number): void
	submitButtonId?: string
}
/**
 *
 * Multiple step form container. `TEMultiStepForm` provides a quick and easy way to break a form into multiple parts and present the user with one part at a time.
 *
 */
export const TEMultiStepForm = React.forwardRef<HTMLFormElement, TEMultiStepFormProps>(
	(props, ref) => {
		const {
			className = '',
			handleCancelOnClick,
			stepData,
			// roundedButtons,
			submitButtonId,
			onStepChange,
			onSubmit,
		} = props
		const stepCount = stepData?.length ?? 0

		const [currentStep, setCurrentStep] = useState(0)

		const updateAndEmitStep = (step: number) => {
			setCurrentStep(step)
			if (onStepChange) {
				onStepChange(step)
			}
		}
		const handlePreviousPressed = async () => {
			const moveBack = () => {
				if (currentStep === 0) {
					if (handleCancelOnClick) {
						handleCancelOnClick()
					}
				} else {
					updateAndEmitStep(currentStep - 1)
				}
			}

			try {
				const onPrevious = stepData && stepData?.[currentStep]?.onPrevious
				if (onPrevious) {
					await onPrevious()
					moveBack()
				} else {
					moveBack()
				}
			} catch (e) {
				console.log(e)
			}
		}
		const handleNextPressed = async () => {
			const moveForward = async () => {
				try {
					if (currentStep + 1 === stepData.length) {
						await onSubmit()
						updateAndEmitStep(0)
					} else {
						updateAndEmitStep(currentStep + 1)
					}
				} catch (e) {
					console.log(e)
				}
			}
			try {
				const onNext = stepData && stepData?.[currentStep]?.onNext
				if (onNext) {
					await onNext()
					moveForward()
				} else {
					moveForward()
				}
			} catch (e) {
				console.log(e)
			}
		}

		return (
			<Container className={`TEMultiStepForm ${className}`}>
				<StepContainer className='TEMultiStepFormStepContainer'>
					<StepBar stepCount={stepCount} className='TEMultiStepFormStepBar' />
					{stepData &&
						stepData.map((step, index) => {
							const { title } = step

							return (
								<StepWrapper
									key={index}
									stepCount={stepCount}
									className='TEMultiStepFormStepWrapper'
								>
									<StepNumber
										active={currentStep === index}
										className='TEMultiStepFormStepNumber'
									>
										{index + 1}
									</StepNumber>
									<StepTitle className='TEMultiStepFormStepTitle'>
										{title}
									</StepTitle>
								</StepWrapper>
							)
						})}
				</StepContainer>
				<TEForm className='TEMultiStepFormForm' ref={ref}>
					{stepData && stepData.length > 0 && stepData?.[currentStep]?.component}
					<ButtonContainer className='TEMultiStepFormButtonContainer'>
						{(handleCancelOnClick || currentStep > 0) && (
							<StepButton
								onClick={handlePreviousPressed}
								// position="left"
								// rounded={roundedButtons}
								className='TEMultiStepFormStepButton TEMultiStepFormStepButtonLeft'
							>
								{currentStep === 0 ? 'cancel' : 'previous'}
							</StepButton>
						)}
						<StepButton
							onClick={handleNextPressed}
							// position="right"
							// rounded={roundedButtons}
							// singleButton={!handleCancelOnClick && currentStep === 0}
							className='TEMultiStepFormStepButton TEMultiStepFormStepButtonright'
							id={submitButtonId}
						>
							{currentStep + 1 === stepCount ? 'submit' : 'next'}
						</StepButton>
					</ButtonContainer>
				</TEForm>
			</Container>
		)
	},
)
