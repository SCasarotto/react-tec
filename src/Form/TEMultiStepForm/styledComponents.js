import styled from 'styled-components'

import TEForm from './../TEForm'
import TEButton from './../TEButton'

export const Container = styled.div`
	${(props) => {
		return `
        `
	}}
`
export const StepContainer = styled.div`
	${(props) => {
		return `
            width: 100%;
            display: flex;
            position: relative;
            margin-bottom: 30px;

            @media (max-width: 500px) {
                margin-bottom: 15;
            }
        `
	}}
`
export const StepBar = styled.div`
	${(props) => {
		const { stepCount, theme } = props
		return `
            position: absolute;
            top: 14px;
            background-color: ${theme.primary};
            height: 2px;
            width: ${100 - 100 / stepCount}%;
            left: ${100 / stepCount / 2}%;
        `
	}}
`
export const StepWrapper = styled.div`
	${(props) => {
		const { stepCount } = props
		return `
            display: flex;
            flex-direction: column;
            align-items: center;
            width: ${100 / stepCount}%
        `
	}}
`
export const StepNumber = styled.div`
	${(props) => {
		const { active, theme } = props
		return `
            text-align: center;
            width: 30px;
            height: 30px;
            margin-bottom: 10px;
            line-height: 30px;
            border: 1px solid ${theme.primary};
            z-index: 1;
            background-color: ${active ? theme.primary : theme.white};
            color: ${active ? theme.white : theme.primary};
            transition: color 0.2s ease-in, background-color 0.2s ease-in;
        `
	}}
`
export const StepTitle = styled.span`
	${(props) => {
		return `
            text-align: center;
            font-size: 18px;
            display: block;

            @media (max-width: 500px) {
                font-size: 16px;
            }
            @media (max-width: 350px) {
                font-size: 14px;
            }
        `
	}}
`

export const Form = styled(TEForm)`
	${(props) => {
		return `
            margin-bottom: 30px;
        `
	}}
`
export const ButtonContainer = styled.div`
	${(props) => {
		return `
            display: flex;
        `
	}}
`

//TODO: Return to this and figure out way to handle popups and panel use cases without being weird to use in general
export const StepButton = styled(TEButton)`
	${(props) => {
		// const { position, rounded, singleButton, theme } = props
		// let styles = `
		//     padding: 10px 20px;
		//     color: ${theme.primary};
		//     background-color: ${theme.white};
		//     border-top: 1px solid ${theme.primary};
		//     border-bottom: none;
		//     transition: border 0.2s ease-in, color 0.2s ease-in, background-color 0.2s ease-in;
		// `

		// switch (position) {
		//     case 'left':
		//         styles += `
		//                 border-left: none;
		//                 border-right: 1px solid ${theme.primary};

		//                 border-bottom-left-radius: ${rounded ? '5px' : '0px'};
		//                 border-top-right-radius: 0px;
		//                 border-top-left-radius: 0px;
		//                 border-bottom-right-radius: 0px;

		//                 :hover {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: none;
		//                     border-right: 1px solid ${theme.primary};
		//                     border-bottom: none;
		//                 }
		//                 :active {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: none;
		//                     border-right: 1px solid ${theme.primary};
		//                     border-bottom: none;
		//                 }
		//             `
		//         break
		//     case 'right':
		//         styles += `
		//                 border-left: ${singleButton ? 'none' : `1px solid ${theme.primary}`};
		//                 border-right: none;

		//                 border-bottom-right-radius: ${rounded ? '5px' : '0px'};
		//                 border-bottom-left-radius: ${singleButton && rounded ? '5px' : '0px'};
		//                 border-top-right-radius: 0px;
		//                 border-top-left-radius: 0px;

		//                 :hover {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: ${singleButton ? 'none' : `1px solid ${theme.primary}`};
		//                     border-right: none;
		//                     border-bottom: none;
		//                 }
		//                 :active {
		//                     color: ${theme.white};
		//                     background-color: ${theme.primary};
		//                     border-top: 1px solid ${theme.primary};
		//                     border-left: ${singleButton ? 'none' : `1px solid ${theme.primary}`};
		//                     border-right: none;
		//                     border-bottom: none;
		//                 }
		//             `
		//         break
		//     default:
		//         break
		// }

		return ''
	}}
`
