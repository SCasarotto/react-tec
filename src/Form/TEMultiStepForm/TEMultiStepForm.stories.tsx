import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEMultiStepForm, TEMultiStepFormProps } from './TEMultiStepForm'

export default {
	title: 'Form/TEMultiStepForm',
	component: TEMultiStepForm,
	argTypes: {},
} as Meta

const Template: Story<TEMultiStepFormProps> = (args) => <TEMultiStepForm {...args} />

export const Default = Template.bind({})
Default.args = {}
