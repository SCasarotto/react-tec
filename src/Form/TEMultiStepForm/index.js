import TEMultiStepForm from './TEMultiStepForm'

export default TEMultiStepForm
