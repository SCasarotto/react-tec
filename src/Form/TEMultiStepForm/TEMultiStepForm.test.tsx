import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { TEMultiStepForm } from './TEMultiStepForm'

describe('TEMultiStepForm', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TEMultiStepForm stepData={[]} onSubmit={async () => {}} />)
		expect(asFragment()).toMatchSnapshot()
	})

	// Class Name
	it('accepts custom className', () => {
		const { container } = render(
			<TEMultiStepForm stepData={[]} onSubmit={async () => {}} className='myClassName' />,
		)
		expect(container.firstChild).toHaveClass('myClassName')
	})

	// Ref
	it('renders with ref', async () => {
		const ref: { current: HTMLFormElement | null } = { current: null }
		render(
			<TEMultiStepForm
				stepData={[]}
				onSubmit={async () => {}}
				ref={ref}
				data-testid='myMultiForm'
			/>,
		)
		// Using this as the focus check didn't seem to work
		waitFor(() => expect(ref.current).toBeTruthy())
	})
})
