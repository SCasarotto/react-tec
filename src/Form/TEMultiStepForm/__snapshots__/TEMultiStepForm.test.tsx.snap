// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEMultiStepForm matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  position: relative;
  width: 100%;
  margin: 0px auto;
  box-sizing: border-box;
}

.c4 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 18px;
  padding: 5px 20px;
  text-align: center;
  cursor: pointer;
  border-radius: 5px;
  box-sizing: border-box;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: color 0.2s ease-in,background-color 0.2s ease-in,border 0.2s ease-in;
  transition: color 0.2s ease-in,background-color 0.2s ease-in,border 0.2s ease-in;
}

.c4:hover,
.c4:active {
  border: 1px solid;
}

.c4:disabled {
  cursor: not-allowed;
  border: 1px solid;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin-bottom: 30px;
}

.c1 {
  position: absolute;
  top: 14px;
  height: 2px;
  width: -Infinity%;
  left: Infinity%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:500px) {
  .c0 {
    margin-bottom: 15;
  }
}

<div
    class="TEMultiStepForm "
  >
    <div
      class="c0 TEMultiStepFormStepContainer"
    >
      <div
        class="c1 TEMultiStepFormStepBar"
      />
    </div>
    <form
      class="c2 TEForm TEMultiStepFormForm"
    >
      <div
        class="c3 TEMultiStepFormButtonContainer"
      >
        <button
          class="c4 TEButton TEMultiStepFormStepButton TEMultiStepFormStepButtonright"
          type="button"
        >
          next
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
